clients/orderbook/src/tests.rs

```diff
@@ -0,0 +1,710 @@
+		main: alice_acc.clone(),
+		proxy: bob_acc,
+	};
+
+	worker.handle_blk_import(0).unwrap();
+
+	worker.process_withdraw(withdraw_request, 0).unwrap();
+	let charlie = AccountKeyring::Charlie.pair();
+	let charlie_acc = AccountId::from(charlie.public());
+	let withdraw_request = WithdrawalRequest {
+		signature: charlie.sign(&payload.encode()).into(),
+		payload,
+		main: alice_acc,
+		proxy: charlie_acc,
+	};
+	worker.process_withdraw(withdraw_request, 0).unwrap()
+}
+
+// Setup runtime
+#[tokio::test]
+pub async fn test_offline_storage() {
+	let alice = AccountKeyring::Alice.pair();
+	let bob = AccountKeyring::Bob.pair();
+	let alice_acc = AccountId::from(alice.public());
+	let bob_acc = AccountId::from(bob.public());
+	create_test_api!(
+		one_validator,
+		latest_summary: SnapshotSummary::default(),
+		ingress_messages:
+			vec![
+				IngressMessages::RegisterUser(
+					AccountId::from(AccountKeyring::Alice.pair().public()),
+					AccountId::from(AccountKeyring::Bob.pair().public())
+				),
+				IngressMessages::AddProxy(
+					AccountId::from(AccountKeyring::Alice.pair().public()),
+					AccountId::from(AccountKeyring::Charlie.pair().public())
+				),
+				IngressMessages::Deposit(
+					AccountId::from(AccountKeyring::Alice.pair().public()),
+					AssetId::polkadex,
+					Decimal::from_f64(10.2).unwrap()
+				)
+			],
+		AccountKeyring::Alice
+	);
+	let api = Arc::new(one_validator::TestApi {});
+	// Setup worker
+	let testnet = ObTestnet::new(1, 0);
+	let peer = &testnet.peers[0];
+	let (rpc_sender, rpc_receiver) = futures::channel::mpsc::unbounded();
+
+	let worker_params = WorkerParams {
+		client: peer.client().as_client(),
+		backend: peer.client().as_backend(),
+		runtime: api,
+		sync_oracle: peer.network_service().clone(),
+		network: peer.network_service().clone(),
+		protocol_name: Cow::from("blah"),
+		is_validator: true,
+		message_sender_link: rpc_receiver,
+		metrics: None,
+		_marker: Default::default(),
+		last_successful_block_number_snapshot_created: Arc::new(RwLock::new(
+			0_u32.saturated_into(),
+		)),
+		memory_db: Arc::new(RwLock::new(MemoryDB::default())),
+		working_state_root: Arc::new(RwLock::new([0; 32])),
+	};
+	assert!(worker_params.backend.offchain_storage().is_some());
+	let mut worker = ObWorker::new(worker_params);
+}
+
+#[test]
+pub fn test_trie_insertion() {
+	let mut working_state_root = [0u8; 32];
+	let mut memory_db: MemoryDB<RefHasher, HashKey<RefHasher>, Vec<u8>> = Default::default();
+	{
+		let mut trie: TrieDBMut<ExtensionLayout> =
+			TrieDBMutBuilder::new(&mut memory_db, &mut working_state_root).build();
+
+		//trie.insert(b"ab".as_ref(),b"cd".as_ref()).unwrap();
+		trie.commit();
+	}
+
+	// {
+	//     let mut trie: TrieDBMut<ExtensionLayout> =
+	//         TrieDBMutBuilder::from_existing(&mut memory_db, &mut working_state_root)
+	//             .build();
+	//    assert_eq!(trie.get(b"ab".as_ref()).unwrap(), Some(b"cd".to_vec()))
+	// }
+
+	assert_ne!(working_state_root, [0u8; 32]);
```

Comment:
Should be properly tested with expected real value and not just that it's different from the initial...

Plus all commented out code should be removed